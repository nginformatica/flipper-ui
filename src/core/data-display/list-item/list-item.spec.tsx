import React from 'react'
import { render, screen } from '@testing-library/react'
import ListItem from '.'

describe('ListItem', () => {
    it('should render', () => {
        render(<ListItem title='Item 1' />)

        const listItem = screen.getByText('Item 1')

        expect(listItem).toBeDefined()
    })

    it('should render with avatar', () => {
        render(<ListItem title='Item 1' avatar={<div>Avatar</div>} />)

        const avatar = screen.getByText('Avatar')

        expect(avatar).toBeDefined()
    })

    it('should render with action', () => {
        render(<ListItem title='Item 1' action={<div>Action</div>} />)

        const action = screen.getByText('Action')

        expect(action).toBeDefined()
    })

    it('should match snapshot', () => {
        const { container } = render(
            <ListItem
                title='Item 1'
                avatar={<div>Avatar</div>}
                action={<div>Action</div>}
            />
        )

        expect(container).toMatchSnapshot()
    })
})
