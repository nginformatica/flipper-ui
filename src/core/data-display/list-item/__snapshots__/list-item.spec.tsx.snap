// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should match snapshot 1`] = `
<div>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root makeStyles-root-7 MuiListItem-gutters MuiListItem-button"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemAvatar-root"
    >
      <div>
        Avatar
      </div>
    </div>
    <div
      class="MuiListItemText-root"
      data-testid="list-item-undefined"
      style="margin-right: 36px;"
    >
      <span
        class="MuiTypography-root makeStyles-default-8 MuiTypography-body1 MuiTypography-displayBlock"
      >
        Item 1
      </span>
    </div>
    <div
      class="MuiListItemSecondaryAction-root makeStyles-default-8"
    >
      <div>
        Action
      </div>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</div>
`;
